apps/docs/components/ExampleCodeBlock.tsx

```diff
@@ -0,0 +1,51 @@
+'use client'
+
+import { SandpackCodeEditor, SandpackFiles, SandpackProvider } from '@codesandbox/sandpack-react'
+import { useTheme } from 'next-themes'
+import { useEffect, useState } from 'react'
+
+export default function ExampleCodeBlock({
+	articleId,
+	files = {},
+	activeFile,
+}: {
+	articleId: string
+	activeFile: string
+	files: SandpackFiles
+}) {
+	const [isClientSide, setIsClientSide] = useState(false)
+	const { theme } = useTheme()
+	useEffect(() => setIsClientSide(true), [])
```

Comment:
Is this to get around some hydration weirdness?

Reply:
yeah, basically - i'll add a comment, now that you mention it 👍